src/ert/gui/model/node.py

```diff
@@ -2,105 +2,68 @@
 
 from abc import ABC, abstractmethod
 from dataclasses import dataclass, field
-from enum import Enum, auto
-from typing import Any, Optional
+from typing import Optional, cast
 
 from qtpy.QtGui import QColor
 
 from ert.ensemble_evaluator.snapshot import ForwardModel
 
 
-class NodeType(Enum):
-    ROOT = auto()
-    ITER = auto()
-    REAL = auto()
-    JOB = auto()
-
-
 @dataclass
 class _Node(ABC):
-    id_: int
-    data: dict[Any, Any] = field(default_factory=dict)
+    id_: str
     parent: Optional[RootNode | IterNode | RealNode] = None
-    children: dict[int, IterNode | RealNode | ForwardModelStepNode] = field(
-        default_factory=dict
-    )
+    children: (
+        dict[str, IterNode] | dict[str, RealNode] | dict[str, ForwardModelStepNode]
+    ) = field(default_factory=dict)
 
     def __repr__(self) -> str:
         parent = "no " if self.parent is None else ""
         children = "no " if len(self.children) == 0 else f"{len(self.children)} "
         return f"Node<{type(self).__name__}>@{self.id_} with {parent}parent and {children}children"
 
     @abstractmethod
-    def add_child(
-        self,
-        node: IterNode | RealNode | ForwardModelStepNode,
-        node_id: Optional[int] = None,
-    ) -> None:
+    def add_child(self, node: _Node) -> None:
```

Comment:
Is it worth changing the node type from `IterNode | RealNode | ForwardModelStepNode` to `_Node`, if we have to cast the types in every method implementation?

Reply:
it will be more or less the same. the signature will be incorrect either way. each node type only accepts one type of child, but you should not change the signature of an overridden method.